kart/base_dataset.py

```diff
@@ -189,12 +191,55 @@ def diff_to_working_copy(
         # Subclasses to override.
         pass
 
+    def get_deltas_from_git_diff_for_subtree(
+        self,
+        other: Optional["BaseDataset"],
+        subtree_name: str,
+        *,
+        reverse: bool = False,
+        path_transform: Callable[[str], str] | None = None,
+    ) -> Iterator[RawDiffDelta]:
+        self_subtree = self.get_subtree(subtree_name)
+        other_subtree = other.get_subtree(subtree_name) if other else self._empty_tree
+
+        if reverse:
+            self_subtree, other_subtree = other_subtree, self_subtree
+
+        # TODO: iterate over 0-terminated 'lines' of output rather than using check_output. Efficiently. How?
```

Comment:
Found this PEP from 2005 that is.... languishing...
It does have a little snippet inside it that shows how you can do this kind of thing though:
https://bugs.python.org/issue1152248